arangod/Pregel/Worker/Worker.cpp

```diff
@@ -566,13 +566,31 @@ void Worker<V, E, M>::finalizeExecution(FinalizeExecution const& msg,
         [this, self = shared_from_this(),
          statusUpdateCallback = std::move(_makeStatusCallback()),
          cleanup = std::move(cleanup)] {
+          // TODO: This try/catch block should be removed. Use Results instead.
           try {
             auto storer = GraphStorer<V, E>(_config, _algorithm->inputFormat(),
                                             _config->globalShardIDs(),
                                             std::move(statusUpdateCallback));
             _feature.metrics()->pregelWorkersStoringNumber->fetch_add(1);
-            // TODO GORDO-1599
-            storer.store(_quivers);
+
+            std::vector<futures::Future<Result>> storeFutures;
+            for (auto const& quiver : _quivers) {
+              storeFutures.emplace_back(storer.store({quiver}));
+            }
+            if (!storeFutures.empty()) {
+              // wait for all futures to finalize
+              auto futureResponses = futures::collectAll(storeFutures).get();
+              for (auto const& fResponse : futureResponses) {
+                auto& result = fResponse.get();
+                if (result.fail()) {
+                  LOG_PREGEL("ee2af", WARN)
+                      << fmt::format("Thread failed to store graph results: {}",
+                                     result.errorMessage());
+                  // TODO: Anything else we can do in case we've failed here?!
+                  throw;
```

Comment:
throw what? `throw;` which is written below make sense only in catch block.
It's rethrow